### PR TITLE
🚨🚨 Source JIRA: Save state for Boards Issues per board

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolving https://github.com/airbytehq/oncall/issues/3757

## How

Save State for every `board` in stream `Boards Issues`

## Recommended reading order
1. `streams.py`

## 🚨 User Impact 🚨

state has changes for stream `Boards Issues`

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-jira/metadata.yaml [hunk 1]
```diff
@@ -24,6 +24,14 @@ data:
     oss:
       enabled: true
   releaseStage: generally_available
+  releases:
+    breakingChanges:
+      1.0.0:
+        message: "Stream State will be saved for every board in stream `Boards Issues`. Customers who use stream `Board Issues` in Incremental sync mode must take action with their connections."
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-jira/metadata.yaml:30]: Stream state will be saved for every board in stream `Boards Issues`. Customers who use stream `Board Issues` in Incremental Sync mode must take action with their connections.

File: docs/integrations/sources/jira-migrations.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Jira Migration Guide
+
+## Upgrading to 1.0.0
+
+Note: this change is only breaking if you are using the `Boards Issues` stream in Incremental Mode.
```
Comment (katmarkham) [inline docs/integrations/sources/jira-migrations.md:5]: Incremental Sync mode

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-jira/metadata.yaml (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-jira/metadata.yaml]: updated

File: docs/integrations/sources/jira-migrations.md (hunk 2 above)
Comment (artem1205) [reply docs/integrations/sources/jira-migrations.md]: updated

**Review by katmarkham — APPROVED** [review]

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 3]
```diff
@@ -288,17 +296,45 @@ def should_retry(self, response: requests.Response) -> bool:
...
+        board_id = str(stream_slice["board_id"])
+        if self.cursor_field not in self._starting_point_cache:
+            self._starting_point_cache.setdefault(board_id, {})[self.cursor_field] = self._get_starting_point(
+                stream_state=stream_state, stream_slice=stream_slice
+            )
+        return self._starting_point_cache[board_id][self.cursor_field]
+
+    def _get_starting_point(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any]) -> Optional[pendulum.DateTime]:
+        board_id = str(stream_slice["board_id"])
+        if stream_state:
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:332]: you don't need assign board_id before if statement. It's used only `stream_state` not false.

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 4]
```diff
@@ -288,17 +296,45 @@ def should_retry(self, response: requests.Response) -> bool:
         # for all other HTTP errors the default handling is applied
         return super().should_retry(response)
 
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
+        self._starting_point_cache.clear()
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:300]: please add comment why we need do this

**Review by lazebnyi — APPROVED** [review]
lgtm 

need small changes

**Pull request merged**